### ISSUE TITLE
CMake build errors

### ISSUE THREAD

**Issue by kcweicht** [Qartar/qflags#1]
* When QFLAGS_BUILD_SAMPLES is OFF, add_custom_command at https://github.com/Qartar/qflags/blob/master/CMakeLists.txt#L217 still references trivial even though that project was never created.
* find_package(googletest) is called even when QFLAGS_BUILD_TESTS is disabled

**Comment by kcweicht** [comment]
Here's the warning from CMake with regards to googletest:

```
CMake Warning at 3rd_party/qflags/CMakeLists.txt:26 (find_package):
  By not providing "Findgoogletest.cmake" in CMAKE_MODULE_PATH this project
  has asked CMake to find a package configuration file provided by
  "googletest", but CMake did not find one.

  Could not find a package configuration file provided by "googletest" with
  any of the following names:

    googletestConfig.cmake
    googletest-config.cmake

  Add the installation prefix of "googletest" to CMAKE_PREFIX_PATH or set
  "googletest_DIR" to a directory containing one of the above files.  If
  "googletest" provides a separate development package or SDK, be sure it has
  been installed.
```

**Comment by Qartar** [comment]
The first issue was fixed in e9121b4395fd5f62aa991f00b8f2131e7dccf8e2. As for the second, it looks like `find_package` was generating a warning because `CMAKE_MODULE_PATH` was being set incorrectly when qflags is included as a subproject which then prevents CMake from finding the package configuration file. Since googletest is not a required dependency it should not generate any warnings when missing after the fix in 86c44de2bab6fd7bec2c7ead4c7ed3cfe93baee6.
